arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -33,104 +190,356 @@ class CommonStressorParams:
+        ),
+    ] = MatrixMethod.ALL
+
+    matrix_size: typing.Annotated[
+        typing.Optional[int],
+        schema.id("matrix-size"),
+        schema.name("Matrix Size"),
+        schema.description("Specify the N x N size of the matrices"),
+    ] = None
+
+    matrix_yx: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("matrix-yx"),
+        schema.name("Matrix YX"),
+        schema.description(
+            "Perform matrix operations in order y by x rather than the "
+            "default x by y"
+        ),
+    ] = None
+
     def to_jobfile(self) -> str:
-        matrix = "matrix {}\n".format(self.workers)
-        result = matrix
+        result = f"matrix {self.workers}\n"
+        if self.matrix_ops is not None:
+            result += f"matrix-ops {self.matrix_ops}\n"
+        if self.matrix_method is not None:
+            result += f"matrix-method {self.matrix_method}\n"
+        if self.matrix_size is not None:
+            result += f"matrix-size {self.matrix_size}\n"
+        if self.matrix_yx is True:
+            result += "matrix-yx\n"
         return result
 
 
 @dataclass
 class MqStressorParams(CommonStressorParams):
+    mq_ops: typing.Annotated[
+        typing.Optional[int],
+        schema.id("mq-ops"),
+        schema.name("MQ Operations"),
+        schema.description(
+            "Stop after N bogo POSIX message send operations completed"
+        ),
+    ] = None
+
+    mq_size: typing.Annotated[
+        typing.Optional[int],
+        schema.id("mq-size"),
+        schema.name("MQ Size"),
+        schema.description(
+            "Specify size of POSIX message queue; the default size is "
+            "10 messages and most Linux systems this is the maximum allowed "
+            "size for normal users"
+        ),
+    ] = None
+
     def to_jobfile(self) -> str:
-        mq = "mq {}\n".format(self.workers)
-        result = mq
+        result = f"mq {self.workers}\n"
+        if self.mq_ops is not None:
+            result += f"mq-ops {self.mq_ops}\n"
+        if self.mq_size is not None:
+            result += f"mq-size {self.mq_size}\n"
         return result
 
 
+@dataclass
+class HDDOptsParams:
+    hdd_opts: typing.Annotated[
+        typing.Optional[HddOpts],
+        schema.name("HDD Options"),
+        schema.description("HDD stress test options"),
+    ] = None
+
+
 @dataclass
 class HDDStressorParams(CommonStressorParams):
     hdd_bytes: typing.Annotated[
-        str,
+        typing.Optional[str],
+        schema.id("hdd-bytes"),
         schema.name("Bytes Per Worker"),
-        schema.description(
-            "write  N  bytes for each hdd process, the default is 1 GB. "
-            "One can specify the size in units of Bytes, KBytes, "
-            "MBytes and GBytes using the suffix b, k, m or g."
-        ),
-    ]
+        schema.description("Write N bytes for each hdd process, the default is 1 GB"),
+    ] = None
+
+    hdd_opts: typing.Annotated[
```

Comment:
Unless there is a compelling reason to the contrary, I recommend renaming this attribute (to something like `hdd_options`):  since there is already another field `hdd_ops` which differs by only one letter in a bad position, these fields are likely to be confused for each other.  Since there are various other "ops" fields, it makes sense to rename this attribute and keep the consistency with the others.

Reply:
There is a compelling reason -- The parameter names map exactly to the stress-ng input parameters (well, with s/-/_/g). This one is just a little unfortunate.